core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/store/dot-palette.store.ts

```diff
@@ -168,11 +172,22 @@ export class DotPaletteStore extends ComponentStore<DotPaletteState> {
             filter: '',
             languageId: '1',
             totalRecords: 0,
-            viewContentlet: 'contentlet:out',
+            viewContentlet: CONTENTLET_VIEW_OUT,
             loading: false
         });
     }
 
+    /**
+     * Switch language and request Content Types data.
+     * @param languageId
+     *
+     * @memberof DotPaletteStore
+     */
+    switchLanguage(languageId: string): void {
+        this.setLanguage(languageId);
+        this.getContenttypesData();
```

Comment:
What this method is doing doesn't look like something that belongs to a store, looks very very anti pattern.